### PR TITLE
Currency added to create new order form

### PR DESCRIPTION
**Pull request by bartoszpietrzak1994**
screen shot 2018-09-03 at 16 04 12

### CODE REVIEW

File: src/Form/Type/CurrencyCodeChoiceType.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    {
+        $this->currencyRepository = $currencyRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $builder->addModelTransformer(new ReversedTransformer(new ResourceToIdentifierTransformer($this->currencyRepository, 'code')));
```
Comment (lchrusciel) [inline src/Form/Type/CurrencyCodeChoiceType.php]: Can we make it in multiple lines?

File: src/Form/Type/NewOrderType.php [hunk 2]
```diff
@@ -75,6 +75,12 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                         'choices' => $event->getData()->getChannel()->getLocales(),
                         'empty_data' => $event->getData()->getLocaleCode(),
                     ])
+                    ->add('currencyCode', CurrencyCodeChoiceType::class, [
+                        'label' => false,
+                        'placeholder' => 'sylius.ui.currency',
+                        'choices' => $event->getData()->getChannel()->getCurrencies(),
```
Comment (lchrusciel) [inline src/Form/Type/NewOrderType.php]: We should at least try to reduce the number of dzidas on each line. Can you extract `$event->getData()` and add typehint?

File: src/Form/Type/NewOrderType.php [hunk 3]
```diff
@@ -75,6 +75,12 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                         'choices' => $event->getData()->getChannel()->getLocales(),
                         'empty_data' => $event->getData()->getLocaleCode(),
                     ])
+                    ->add('currencyCode', CurrencyCodeChoiceType::class, [
+                        'label' => false,
+                        'placeholder' => 'sylius.ui.currency',
```
Comment (lchrusciel) [inline src/Form/Type/NewOrderType.php]: Is there any need for a placeholder, as we have empty_data set anyway?

File: src/Form/Type/CurrencyCodeChoiceType.php (hunk 1 above)
Comment (lchrusciel) [inline src/Form/Type/CurrencyCodeChoiceType.php]: Can we split it to multiple lines?

File: src/Form/Type/NewOrderType.php (hunk 2 above)
Comment (lchrusciel) [inline src/Form/Type/NewOrderType.php]: Can we extract `$event->getData()` to variable and add typehint to it?

File: src/Form/Type/CurrencyCodeChoiceType.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\ReversedTransformer;
+
+final class CurrencyCodeChoiceType extends AbstractType
+{
+    /**
+     * @var RepositoryInterface
+     */
```
Comment (Zales0123) [inline src/Form/Type/CurrencyCodeChoiceType.php]: `/** @var RepositoryInterface */`?

File: src/Form/Type/CurrencyCodeChoiceType.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+     * {@inheritdoc}
+     */
+    public function getParent(): string
+    {
+        return CurrencyChoiceType::class;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
```
Comment (Zales0123) [inline src/Form/Type/CurrencyCodeChoiceType.php]: I don't think we need these `{@inheritdoc}` comments

File: src/Form/Type/NewOrderType.php [hunk 6]
```diff
@@ -71,9 +79,13 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
-                        'placeholder' => 'sylius.ui.locale',
-                        'choices' => $event->getData()->getChannel()->getLocales(),
-                        'empty_data' => $event->getData()->getLocaleCode(),
+                        'choices' => $channel->getLocales(),
+                        'empty_data' => $order->getLocaleCode(),
+                    ])
+                    ->add('currencyCode', CurrencyCodeChoiceType::class, [
+                        'label' => false,
+                        'choices' => $channel->getCurrencies(),
+                        'empty_data' => $order->getCurrencyCode()
```
Comment (Zales0123) [inline src/Form/Type/NewOrderType.php]: Missing comma at the end of line

**Comment by Zales0123** [comment]
🎉

**Pull request merged**